File: Domain-Module/User-Module/User-Domain/src/main/java/com/pawith/userdomain/utils/UserUtils.java
package com.pawith.userdomain.utils;

import com.pawith.commonmodule.util.SecurityUtils;
import com.pawith.userdomain.entity.User;
import com.pawith.userdomain.service.user.UserReader;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

@Slf4j
@Component
@RequiredArgsConstructor
public class UserUtils {

    private final UserReader userReader;

    public User getAccessUser(){
        final Long userId = SecurityUtils.getAuthenticationPrincipal();
        return userReader.findById(userId);
    }

    public Long getIdFromAccessUser(){
        return SecurityUtils.getAuthenticationPrincipal();
    }
}

File: Domain-Module/User-Module/User-Domain/src/main/java/com/pawith/userdomain/UserUtils.kt
package com.pawith.userdomain

import com.pawith.commonmodule.util.SecurityUtils
import com.pawith.userdomain.service.UserReader
import org.springframework.stereotype.Component

@Component
class UserUtils(
    val userReader: UserReader
) {
    fun getAccessUser(): User{
        val userId = SecurityUtils.getAuthenticationPrincipal()
        return userReader.readByUserId(userId)
    }

    fun getIdFromAccessUser() : Long{
        return SecurityUtils.getAuthenticationPrincipal()
    }
}